src/overlays/actors/ovl_Obj_HsStump/z_obj_hsstump.h

```diff
@@ -3,11 +3,21 @@
 
 #include <global.h>
 
+#define OBJHSSTUMP_GET_SWITCHFLAG(this) (((ObjHsStump*)this)->dyna.actor.params & 0x7F)
+#define OBJHSSTUMP_GET_ISHIDDEN(this) ((((ObjHsStump*)this)->dyna.actor.params >> 12) & 0xF)
```

Comment:
Is it better to have these macros take in the `Actor *` instead? Seems like that would be consistent with them requiring `thisx`?

```suggestion
#define OBJHSSTUMP_GET_SWITCHFLAG(actor) ((actor)->params & 0x7F)
#define OBJHSSTUMP_GET_ISHIDDEN(actor) ((actor)->params >> 12) & 0xF)
```


(I'm not sure about this, not sure if it should change or not)

Reply:
I think there are arguments for both sides. The reason I chose to cast as `ObjHsStump` even though `thisx` was passed in was that these macros don't really apply to any old actor. They only apply to `ObjHsStump` actors so the cast seems appropriate to specify this condition. Perhaps it is enough to that the names are prefixed with `ObjHsStump`. Any one have any strong opinions about this?

Reply:
Yeah makes sense. I'm fine either way.

One thing that I was worried about these macros is that the `(ObjHsStump *)` cast would hide some type errors? The compiler won't detect that `OBJHSSTUMP_GET_SWITCHFLAG(thisx->params)` is wrong (and would probably crash when ran). 
I guess this isn't a big deal for this project, because we have the matching checksum?

Reply:
Just make the macros take the params itself rather than the actor instance.

Reply:
It makes sense to accept the derived Actor class, but currently no other macros for params do. Does the code still OK when using the updated macro?

Reply:
Code would still OK with these updated Macros
![image](https://user-images.githubusercontent.com/31974442/119588736-4670d880-bd86-11eb-8686-cdc60a5a855e.png)

Reply:
I kind of like Zellll's suggestions of just passing the params in directly though. The we don't have to worry about passing in the correct pointer.

Reply:
Let's get some more opinions, but I'd like to keep these consistent. We have three options:
- Pass in params.
- Pass in the base actor. (Current behaviour)
- Pass in the derived actor.

Reply:
pass in the params, the point of the macro is to extract data out of the params, not out of the actor instance

Reply:
There's no consensus yet.

We're going to go with the current paradigm for consistency: Accept the Actor* base struct pointer.